`PatternBindingDecl::getPatternList()` can in some cases return an empty list, which causes a
segmentation fault when getting the source range. This case needs to be handled for the extractor.

diff --git a/lib/AST/Decl.cpp b/lib/AST/Decl.cpp
index 75b99a22e73..09115678a82 100644
--- a/lib/AST/Decl.cpp
+++ b/lib/AST/Decl.cpp
@@ -1725,7 +1725,10 @@ StringRef PatternBindingEntry::getInitStringRepresentation(
 
 SourceRange PatternBindingDecl::getSourceRange() const {
   SourceLoc startLoc = getStartLoc();
-  SourceLoc endLoc = getPatternList().back().getSourceRange().End;
+  SourceLoc endLoc = startLoc;
+  if (!getPatternList().empty()) {
+    endLoc = getPatternList().back().getSourceRange().End;
+  }
   if (startLoc.isValid() != endLoc.isValid()) return SourceRange();
   return { startLoc, endLoc };
 }
